docs/proposals/mux-logging-service.md

```diff
@@ -0,0 +1,305 @@
+Introduction
+============
+
+Fluentd gets Kubernetes metadata to enrich each container log message with the
+namespace UUID, the pod UUID, and pod labels and annotations.  If you have a
+large number of nodes, this means many connections to the Kubernetes API
+server.  This also requires a lot of processing and filtering on each node.
+
+The component called *mux* is a Service and a DeploymentConfig that allows the
+Fluentd collectors running on each node to offload the Kubernetes metadata
+lookup and other enrichment to a central service.  The Fluentd collectors on
+each node use the *secure_forward* protocol to send the raw logs to *mux* for
+Kubernetes metadata and other enrichment, and then for storage in
+Elasticsearch.  This offloads the processing from each node, and reduces the
+load on the Kubernetes API server.
+
+*mux* is short for multiplex, because it acts like a multiplexor or manifold,
+taking in connections from many collectors, and distributing them to a data
+store.  Another term would be FEAN, short for
+Forward/Enrich/Aggregate/Normalize.
+
+*mux* can also be exposed as an external service.  This allows the ability to
+set up an OpenShift cluster to act as a central logging service for other
+projects and products outside of the cluster.
+
+### Server Side Configuration ###
+
+There is a deploymentConfig - `logging-mux`.  This is the central logging
+service.  There is also a `logging-mux` Service which exposes the
+secure_forward listener port using `externalIPs`.  There is a Secret
+`logging-mux` which holds the server `cert` and `key`, the `ca` cert, and the
```

Comment:
Side note.  Need to refactor this I suspect when client cert generation becomes available to us.